src/drivers/intel/cavs/alh.c

```diff
@@ -29,21 +30,30 @@ static int alh_trigger(struct dai *dai, int cmd, int direction)
 
 static int alh_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 {
+	struct alh_pdata *alh = dai_get_drvdata(dai);
+
 	dai_info(dai, "alh_set_config() config->format = 0x%4x",
 		  config->format);
 
+	alh->params = config->alh;
+
 	return 0;
 }
 
 /* get ALH hw params */
 static int alh_get_hw_params(struct dai *dai,
-			     struct sof_ipc_stream_params  *params, int dir)
+			     struct sof_ipc_stream_params *params, int dir)
 {
+	struct alh_pdata *alh = dai_get_drvdata(dai);
+
+	params->rate = alh->params.rate;
+	params->channels = alh->params.channels;
```

Comment:
check values? e..g channels == 1 or 2, and rate is one of the usual rates.

Reply:
What sort of checks would you expect specifically?